lib/run_action.js

```diff
@@ -6,7 +6,21 @@ var objectToQueryParamString = require('./object_to_query_param_string');
 // This will become require('xhr') in the browser.
 var request = require('request');
 
+/*
+ * "Full Jitter" algorithm taken from https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/
+ */
+function exponentialBackoffWithJitter(numberOfRetries, initialBackoffTimeMs, maxBackoffTimeMs) {
+    var rawBackoffTimeMs = initialBackoffTimeMs * Math.pow(2, numberOfRetries);
+    var clippedBackoffTimeMs = Math.min(maxBackoffTimeMs, rawBackoffTimeMs);
+    var jitteredBackoffTimeMs = Math.random() * clippedBackoffTimeMs;
+    return jitteredBackoffTimeMs;
+}
+
 function runAction(base, method, path, queryParams, bodyData, callback) {
+    return runActionImpl(base, method, path, queryParams, bodyData, callback, 0);
+}
+
+function runActionImpl(base, method, path, queryParams, bodyData, callback, numAttempts) {
```

Comment:
Because we only call `runAction` in one place (from `Base.prototype.runAction`), I'd prefer to make `runAction` take `numAttempts` and call it properply from `Base.prototype.runAction`.

Reply:
👍